zephyr/wrapper.c

```diff
@@ -72,18 +72,32 @@ __section(".heap_mem") static uint8_t __aligned(64) heapmem[HEAPMEM_SIZE];
 
 #else
 
-#define HEAPMEM_SHARED_SIZE	(HEAP_SYSTEM_SIZE + HEAP_RUNTIME_SIZE + \
-				HEAP_RUNTIME_SHARED_SIZE + HEAP_SYSTEM_SHARED_SIZE)
+#if (CONFIG_HP_MEMORY_BANKS <= 8)
+/* APL */
+#define	HEAP_SIZE	0x30000
+#elif (CONFIG_HP_MEMORY_BANKS == 16)
+/* JSL */
+#define	HEAP_SIZE	0x80000
+#elif (CONFIG_HP_MEMORY_BANKS == 30)
+/* TGL-H */
+#define	HEAP_SIZE	0x100000
+#elif (CONFIG_HP_MEMORY_BANKS >= 46)
+/* CNL/ICL/TGL */
+#define	HEAP_SIZE	0x200000
+#endif
+
 #ifdef ENABLE_CACHED_HEAP
-#define HEAPMEM_SIZE		HEAP_BUFFER_SIZE
+/* half heap is cacheable */
+#define	HEAP_SYSTEM_CACHED_SIZE	(HEAP_SIZE / 2)
```

Comment:
Hmm, this is problematic. We can get size of the heap from linker (and hardcode it in shortterm), but the ratio of the heap that should be allocated to coherent-versus-noncoherent usage is SOF specific information and that we'd have to get from SOF.  We should assume "half is cacheable" here, but rather this should come from some common definition.

Reply:
Ack @keyonjie . So yeah, then I think we should follow suite with Zephyr and use a single heap as well and serve both cached and non-cached allocations from the same heap area. No need to block this PR to wait for the Zephyr work to complete.